WorkspaceServer.Tests/Kernel/LanguageKernelTests.cs

```diff
@@ -0,0 +1,1117 @@
+                Language.FSharp => new[]
+                {
+                    "open System",
+                    "aaaadd"
+                },
+
+                Language.CSharp => new[]
+                {
+                    "using System;",
+                    "aaaadd"
+                }
+            };
+
+            await SubmitCode(kernel, source);
+
+            KernelEvents.ValuesOnly()
+                .Where( x => x.GetType() != typeof(KernelIdle) && x.GetType() != typeof(KernelBusy) )
+                .Last()
+                .Should()
+                .BeOfType<CommandFailed>()
+                .Which
+                .Message
+                .Should()
+                .Be("(1,1): error CS0103: The name 'aaaadd' does not exist in the current context");
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        //[InlineData(Language.FSharp)]                     // Todo: FSI should support incomplete submissions  testcase it_cannot_execute_incomplete_submissions
+        public async Task it_can_analyze_incomplete_submissions(Language language)
+        {
+            var kernel = CreateKernel(language);
+
+            var source = language switch
+            {
+                Language.FSharp => "let a =",
+                Language.CSharp => "var a ="
+            };
+
+            await SubmitCode(kernel, source, submissionType: SubmissionType.Diagnose);
+
+            KernelEvents
+                .ValuesOnly()
+                .Single(e => e is IncompleteCodeSubmissionReceived);
+
+            KernelEvents
+                .Should()
+                .Contain(e => e.Value is IncompleteCodeSubmissionReceived);
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        //[InlineData(Language.FSharp)]                     // Todo: FSI - returns a value
+        public async Task it_can_analyze_complete_submissions(Language language)
+        {
+            var kernel = CreateKernel(language);
+
+            var source = language switch
+            {
+                Language.FSharp => "25",
+                Language.CSharp => "25"
+            };
+
+            await SubmitCode(kernel, source, submissionType: SubmissionType.Diagnose);
+
+            KernelEvents
+                .Should()
+                .NotContain(e => e.Value is ReturnValueProduced);
+
+            KernelEvents
+                .Should()
+                .Contain(e => e.Value is CompleteCodeSubmissionReceived);
+        }
+
+        [Theory]
+        [InlineData(Language.CSharp)]
+        //[InlineData(Language.FSharp)]                     // Todo: FSI - returns a value
+        public async Task it_can_analyze_complete_stdio_submissions(Language language)
+        {
+            var kernel = CreateKernel(language);
+
+            var source = language switch
+            {
+                Language.FSharp => "Console.WriteLine(\"Hello\")",
+                Language.CSharp => "Console.WriteLine(\"Hello\")"
```

Comment:
In this case the test will pass for the wrong reason because this is not a complete C# statement unless you add a semicolon to the end.

Reply:
@jonsequitur 
Console.WriteLine("Hello, World") is a complete C# expression, if it was incomplete this would fail:

````
            KernelEvents
                .Should()
                .Contain(e => e.Value is CompleteCodeSubmissionReceived);
````
This is the notebook with a C# kernel note no ';'
![image](https://user-images.githubusercontent.com/5175830/66246801-fc784200-e6cb-11e9-8f8f-1b4e46155eaa.png)